### ISSUE TITLE
Build Failure: Running gx install hangs fetching ipfs objects

### ISSUE THREAD

**Issue by vyzo** [libp2p/go-libp2p#81]
I am trying to build go-libp2p in a fresh workspace. 
After running make, the obvious next step in building seems to be `gx install`.
Alas, it hangs while fetching dependencies from ipfs; running with `--verbose` hangs here:
```
  - fetching QmZ4Qi3GaRbjcx28Sme5eMH7RQjGkt8wHxt2a65oLaeFEV via ipfs api
```

**Comment by vyzo** [comment]
If I C-c the process and restart, it seems to think the package is locally available -- only to hang in a later fetch:
```
  - fetching QmeLQ13LftT9XhNn22piZc3GP56fGqhijuL5Y8KdUaRn1g via ipfs api
```

**Comment by vyzo** [comment]
Also, there don't appear to be any artefacts from these fetches in my go workspace.

**Comment by vyzo** [comment]
Upon further experimentation, it seems that it starts a number of fetches in the background with a non-deterministic process (goroutines problably), resulting in varying fetch print order.
The end result is the same however, it just hangs for a long time

Note: I am not running a local ipfs node.

**Comment by vyzo** [comment]
Running with a local ipfs node seems to be making progress.

Perhaps running ipfs should be a hard dependency listed in the README?

**Comment by vyzo** [comment]
An update: it made solid albeit slow progress, until it got stuck installing go-libp2p-loggables.
It's been stuck there for over an hour now:
```
installing package: go-libp2p-loggables-1.0.7
  - QmRBqJF7hb8ZSpRcMwUt8hNhydWcxGEhtk81HKq6oUwKvs not found locally, fetching into /home/vyzo/src/golang/src/gx/ipfs/QmRBqJF7hb8ZSpRcMwUt8hNhydWcxGEhtk81HKq6oUwKvs
  - fetching QmRBqJF7hb8ZSpRcMwUt8hNhydWcxGEhtk81HKq6oUwKvs via ipfs api
```

**Comment by jbenet** [comment]
@whyrusleeping what's going on there? Missing packages?

**Comment by vyzo** [comment]
There seems to be some IPFS routing issue -- the object seems to be resolvable from other locations.

**Comment by vyzo** [comment]
Restarting the ipfs daemon magically fixed the problem and installation completed!

**Comment by whyrusleeping** [comment]
@vyzo what version of ipfs are you using? This is likely the major bug in 0.4.2 that we've fixed in 0.4.3. If you see this sort of thing somewhat frequently, i would recommend trying out the release candidate for 0.4.3

**Comment by vyzo** [comment]
Aha! I am using ipfs-0.4.2 (it has an ebuild in gentoo, so it was convenient).
